quinn-proto/src/connection/mod.rs

```diff
@@ -3537,21 +3551,35 @@ impl Connection {
     /// frames. Only changes if the length of the remote connection ID changes, which is expected to
     /// be rare.
     fn max_1rtt_overhead(&self) -> usize {
+        // 1 byte for flags, 4 for worst-case packet number
+        1 + self.rem_cids.active().len() + 4 + self.tag_len_1rtt()
+    }
+
+    /// Exact size of non-frame data for the next 1-RTT packet
+    ///
+    /// Packet number size is sensitive to latency and packet loss, so unlike `max_1rtt_overhead`
+    /// this may change unpredictably under normal conditions.
+    fn exact_1rtt_overhead(&self) -> usize {
```

Comment:
Maybe consider using a single function for `exact_1rtt_overhead()` and `max_1rtt_overhead()` with an `Option` argument for `pn`? Reduces duplication and makes the data dependency on the packet number more obvious in the caller.